src/services/inventoryItemStore.ts

```diff
@@ -88,6 +88,12 @@ export async function getInventoryItemsByUserId(
 
     const total = responseBody.hits.total;
     const results = responseBody.hits.hits;
+    const results_remap = results.map((result) => result._source);
```

Comment:
you're actually looping 4 times over all the items:
- once to create a temporary variable that holds the "remapped" results
- once to create a list of all inventory ids
- once to mutate the the first temporary variable with the remapped results
- one more time to create yet another array that holds the same mutated values as the first temporary remapped results

so the first issue is obviously performance, you only need to loop once over all the items to create the desired remapped value
the other issue is that you're actually mutating the original results.

you can technically removed all yours changes, and on line 100 simply do:

```
results.map(({_id, _source }) => { ..._source, id: _id})
```